File: UCMS.Models/DbModels/SiteContentCreation/Site.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UCMS.Models.Interface;
using UCMS.Models.ModelEnums.Site;

namespace UCMS.Models.DbModels.SiteContentCreation
{
    public class Site : IEntity
    {
        public int Id { get => SiteId; }


        public int SiteId { get; set; }
        public string SiteName { get; set; } = string.Empty;
        public string Domain { get; set; } = string.Empty;
        public string Status { get; set; } = string.Empty; // SiteStatus
        public DateTime LastUpdated { get; set; }
        public string? ImageUrl { get; set; }
        public string? ImageAlt { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public ICollection<DocumentType> DocumentTypes { get; set; } = [];
    }
}

File: UCMS.FrontendReactVite/src/models/siteContentCreation/Site.ts
import type { DocumentType } from "./DocumentType";
import { UserDefaultState, type User } from "./User";

export interface Site {
    siteId: number;
    siteName: string;
    domain: string;
    status: "Published" | "Draft" | "Review" | string; // Allow other strings for flexibility
    lastUpdated: Date;
    imageUrl: string;
    imageAlt: string;
    documentTypes : DocumentType[];

    userId: number,
    user: User,
};

export const SiteDefaultState: Site = {
    siteId: 0,
    siteName: '',
    domain: '',
    status: "Draft",
    lastUpdated: new Date(),
    imageUrl: '',
    imageAlt: '',
    documentTypes: [],
    userId: 0,
    user: UserDefaultState,
};

export function makeSiteDependencies(site: Site) : Site {
    site.documentTypes.forEach(doctype => {
        doctype.site = site;

        doctype.contents.forEach(content => {
            content.documentType = doctype;

            content.contentProperties.forEach(contentProperty => {
                contentProperty.content = content;
            })
        });

        doctype.properties.forEach(property => {
            property.documentType = doctype;

            property.contentProperties.forEach(contentProperty => {
                contentProperty.property = property;
            });
        });

        doctype.contents.forEach(content => {
            doctype.properties.forEach(property => {

                content.contentProperties.forEach(contentContentProperty => {

                    property.contentProperties.forEach(propertyContentProperty => {
                        
                        if(contentContentProperty.contentPropertyId == propertyContentProperty.contentPropertyId) {
                            contentContentProperty.property = propertyContentProperty.property;
                            propertyContentProperty.content = contentContentProperty.content;
                        }
                        
                    });

                });

            });
        });
    });
    return site;
}